src/gui/sharee.cpp

```diff
@@ -81,64 +81,26 @@ void ShareeModel::fetch(const QString &search, const ShareeSet &blacklist)
 
 void ShareeModel::shareesFetched(const QJsonDocument &reply)
 {
-    const auto data = reply.object().value("ocs").toObject().value("data").toObject();
-
     QVector<QSharedPointer<Sharee>> newSharees;
 
     /*
      * Todo properly loop all of this
      */
-    const auto exact = data.value("exact").toObject();
     {
-        const auto users = exact.value("users").toArray();
-        foreach (const auto &user, users) {
-            newSharees.append(parseSharee(user.toObject()));
-        }
-
-        const auto groups = exact.value("groups").toArray();
-        foreach (const auto &group, groups) {
-            newSharees.append(parseSharee(group.toObject()));
-        }
 
-        const auto emails = exact.value("emails").toArray();
-        foreach (const auto &email, emails) {
-            newSharees.append(parseSharee(email.toObject()));
-        }
+        const auto appendSharees = [this](const QJsonObject &data, QVector<QSharedPointer<Sharee>>& out) {
+            const QStringList shareeCategoryKeys {"users", "groups", "emails", "remotes", "circles"};
 
-        const auto remotes = exact.value("remotes").toArray();
-        foreach (const auto &remote, remotes) {
-            newSharees.append(parseSharee(remote.toObject()));
-        }
-
-        const auto circles = exact.value("circles").toArray();
-        foreach (const auto &circle, circles) {
-            newSharees.append(parseSharee(circle.toObject()));
-        }
-    }
+            for (const auto &categoryKey : qAsConst(shareeCategoryKeys)) {
+                auto category = data.value(categoryKey).toArray();
```

Comment:
That might require going through `QString::fromUtf8()` here to get the constexpr above you'll likely really on a list of `const char *` instead of a list of `QString`.

Also `category` should be declared const, and then you can drop the `qAsConst` on the following line.